ql/src/java/org/apache/hadoop/hive/ql/parse/TaskCompiler.java

```diff
@@ -472,6 +474,28 @@ private void setLoadFileLocation(
       loc = cmv.getLocation();
     }
     Path location = (loc == null) ? getDefaultCtasLocation(pCtx) : new Path(loc);
+    if (pCtx.getQueryProperties().isCTAS()) {
+      boolean isExternal = pCtx.getCreateTable().isExternal();
+      boolean isAcid = pCtx.getCreateTable().getTblProps().getOrDefault(
+              hive_metastoreConstants.TABLE_IS_TRANSACTIONAL, "false").equalsIgnoreCase("true") ||
+              pCtx.getCreateTable().getTblProps().containsKey(hive_metastoreConstants.TABLE_TRANSACTIONAL_PROPERTIES);
+      if ((HiveConf.getBoolVar(conf, HiveConf.ConfVars.CREATE_TABLE_AS_EXTERNAL) || isExternal) && !isAcid) {
```

Comment:
I think this condition should be removed - let the transformer decide in which case it wants to act

Reply:
Two reasons I don't want to remove this condition.
1) If we can have this condition we can avoid calling the HMS dry run api, if the tables are managed tables (which is the current behavior).
2) Only HS2 would know if hive.create.as.external.legacy config, so this condition should be in HS2
3) L#486 is introducing buckerting_version table parameter in the explain plan for all the tables. Even if I hide it in L#492 this paramter would be missing in the describe/explain table commands. It is failing few tests. 
So I think this should be ok leaving this condition.

Reply:
1. doesnt matter; if that will be a performance bottleneck later we will handle it - but as it is now; it only adds complexity/reduces readability...this stuff works incorrectly because its too complicated already; it would be better to stop add twists...
2. add a parameter to the rpc call and pass the value of `hive.create.as.external.legacy` over to the transformer so it will be able to handle that as well.
3. can't we keep a full table object in the `ctd` instead of just some parts of it? could that help overcome that issue?